tests/monitoring/component_monitoring.go

```diff
@@ -112,44 +113,109 @@ var _ = Describe("[sig-monitoring]Component Monitoring", Serial, decorators.SigM
 
 	Context("Up metrics", func() {
 		BeforeEach(func() {
+			By("Backing up the operator, controller, api and handler scales")
 			scales = libmonitoring.NewScaling(virtClient, []string{virtOperator.deploymentName, virtController.deploymentName, virtApi.deploymentName})
 			scales.UpdateScale(virtOperator.deploymentName, int32(0))
 
+			By("Reducing the alert pending time")
 			libmonitoring.ReduceAlertPendingTime(virtClient)
 		})
 
 		AfterEach(func() {
-			scales.RestoreAllScales()
-			waitUntilComponentsAlertsDoNotExist(virtClient)
+			By("Restoring the operator")
+			oldVirtOperatorReplicas := scales.GetScale(virtOperator.deploymentName)
+
+			for i := int32(1); i <= oldVirtOperatorReplicas; i++ {
+				By(fmt.Sprintf("Updating the operator scale to %d", i))
+				scales.UpdateScale(virtOperator.deploymentName, i)
+
+				By("Waiting for the operator to be up")
+				libmonitoring.WaitForMetricValue(virtClient, "kubevirt_virt_operator_up", float64(i))
+
+				By("Waiting for the operator to be ready")
+				libmonitoring.WaitForMetricValue(virtClient, "kubevirt_virt_operator_ready", float64(i))
+			}
+
+			By("Waiting for an operator to be leading")
+			libmonitoring.WaitForMetricValue(virtClient, "kubevirt_virt_operator_leading", 1.0)
 		})
 
-		It("VirtOperatorDown and NoReadyVirtOperator should be triggered when virt-operator is down", func() {
+		It("VirtOperatorDown should be triggered when virt-operator is down", func() {
```

Comment:
maybe instead of bringing down and up before and after each IT we can use 1 IT and check all metrics and alerts together? to reduce time and flakes?

cc @sradco

Reply:
that's the thing, it doesn't reduce the flakes, it increases them due to more entropy in each test

but would definitely reduce the test time, but if they fail often, doesn't look like an improvement to me